pkg/controller/build/build_controller.go

```diff
@@ -235,67 +233,60 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 
 	// Populate the status struct with default values
 	b.Status.Registered = corev1.ConditionFalse
-	b.Status.Reason = succeedStatus
+	b.Status.Reason = build.SucceedStatus
 
 	// Validate if remote repository exists
```

Comment:
@qu1queee, I second @gabemontero's suggestion, and I also add that lines [238 until 286](https://github.com/shipwright-io/build/blob/1725668c840b654f00f8ad149a9b79608986c302/pkg/controller/build/build_controller.go#L238-L286) should be extracted to a standalone method as well.

As suggested, we can find a common interface and make sure the validation methods are following as such, and therefore create a simple for loop to apply validations.

To summarize, my suggestion is:
1. Extract all validations to a single method;
2. Find a single interface to all validation methods, make sure they follow such interface;
3. Create a slice with all validation methods listed;
4. Loop through the slice, calling the validation method accordingly;

Reply:
@otaviof thanks. I was waiting for your reply based on https://github.com/shipwright-io/build/pull/536#discussion_r564615513 . The majority wants this change, so let me do this. Thanks!